src/distributions/utils.rs

```diff
@@ -159,9 +159,8 @@ mod simd_wmul {
     }
 
     wmul_impl! { (u16x2, u32x2),, 16 }
-    #[cfg(not(target_feature = "sse2"))]
     wmul_impl! { (u16x4, u32x4),, 16 }
-    #[cfg(not(target_feature = "sse4.2"))]
+    #[cfg(not(target_feature = "sse2"))]
     wmul_impl! { (u16x8, u32x8),, 16 }
```

Comment:
IIUC this is a 256-bit op and thus requires AVX?

Reply:
I think packed_simd provides a fallback implementation?

Reply:
Does it?

Can you at least clarify on what basis you decided to make this change?

Reply:
without 128-bit XMM multiplications, packed_simd falls back to a scalar implementation.

Reply:
In that case, why do we bother with these `cfg` selectors at all?

Reply:
I think it is to provide a specialized implementation that uses less instructions.

Reply:
So you're saying the use of a SSE4.2 target originally was a mistake?

Maybe it was, since the insrtuctions appear to be supported on SSE2: https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_mullo_epi16&expand=3967,3990

Reply:
I appears to be a mistake. I guess that was my bad